src/Promitor.Core.Scraping/Configuration/Serialization/v1/Core/MetricDefinitionDeserializer.cs

```diff
@@ -28,9 +29,15 @@ public class MetricDefinitionDeserializer : Deserializer<MetricDefinitionV1>
                 .IsRequired();
             Map(definition => definition.ResourceType)
                 .IsRequired();
+
             Map(definition => definition.AzureMetricConfiguration)
                 .IsRequired()
                 .MapUsingDeserializer(azureMetricConfigurationDeserializer);
+
+            Map(definition => definition.LogAnalyticsConfiguration)
+                // .IsRequired()
```

Comment:
Depending on the outcome of the config proposal discussion, we'd either need to:
- Make them both optional or not
- Introduce validation rule to ensure `AzureMetricConfiguration` is available when non-Log Analytics scrapers are used and vice versa

Reply:
Yeah. I think we can make both optional. And the logic to validate these we put in MetricsValidation.cs
If (resourceType != LogAnalytics) then azureMetricConfiguration field need to be not null
Something like that

Reply:
As mentioned elsewhere, we need to make both optional and add a validation step to ensure either one of them is there based on config. If there is a LA and has monitor config; we might want to log warning saying this will be ignored and vice versa (no need to block)